### PR TITLE
NetSim: Send a single bit when pressing enter in bit-sending levels

### PR DESCRIPTION
**Pull request by islemaster**
Bug found today (not filed):  In early levels like CSP unit 2 stage 4 puzzle 2 you are supposed to send one bit at a time.  Clicking the _Set Wire_ button does correctly, but pressing enter in the textarea would incorrectly send the whole contents as a packet.

Fixes that issue and adds simple tests for the different interaction behaviors.

### CODE REVIEW

File: apps/src/netsim/NetSimPacketEditor.js [hunk 1]
```diff
@@ -478,7 +482,8 @@ var removeWatermark = function (focusEvent) {
  * @returns {boolean} true iff the given event represents a clean enter
  */
 var isUnmodifiedEnterPress = function (jqueryEvent) {
-  return (jqueryEvent.keyCode == 13 && !(jqueryEvent.ctrlKey || jqueryEvent.shiftKey));
+  return (jqueryEvent.keyCode == KeyCodes.ENTER &&
```
Comment (Bjvanminnen) [inline apps/src/netsim/NetSimPacketEditor.js]: Is this intentionally == instead of ===?

File: apps/test/netsim/NetSimSendPanel.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    NetSimTestUtils.initializeGlobalsToDefaultValues();
+    NetSimGlobals.getLevelConfig().defaultEnabledEncodings = ['binary'];
+
+    testShard = NetSimTestUtils.fakeShard();
+    NetSimLocalClientNode.create(testShard, 'Local Lois', function (_, node) {
+      localNode = node;
+    });
+    NetSimLocalClientNode.create(testShard, 'Remote Ralph', function (_, node) {
+      remoteNode = node;
+    });
```
Comment (Bjvanminnen) [inline apps/test/netsim/NetSimSendPanel.js]: I like the names :)

File: apps/test/netsim/NetSimSendPanel.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    it ("sends all packets on pressing enter", function () {
+      panel.packets_[0].setPacketBinary('1000');
+      simulateEnterKeyPress(rootDiv.find('textarea.message'));
+      panel.tick({time:0});
+      assert.equal('', panel.packets_[0].getPacketBinary());
+    });
+  });
+});
```
Comment (Bjvanminnen) [inline apps/test/netsim/NetSimSendPanel.js]: I think our style guide advocates no space after describe/it https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#js-parentheses-spacing

**Comment by Bjvanminnen** [comment]
couple nits, otherwise lgtm

File: apps/test/netsim/NetSimSendPanel.js (hunk 3 above)
Comment (islemaster) [inline apps/test/netsim/NetSimSendPanel.js]: Good call.  I should do a pass to normalize this in NetSim test code.

File: apps/src/netsim/NetSimPacketEditor.js (hunk 1 above)
Comment (islemaster) [inline apps/src/netsim/NetSimPacketEditor.js]: I don't think so!  Didn't even notice when I dropped the constant in; nice catch.  I should update my file-specific jshint settings to warn about these.

Comment (islemaster) [inline apps/src/netsim/NetSimPacketEditor.js]: http://jshint.com/docs/options/#eqeqeq

**Pull request merged**